metricbeat/module/kubernetes/state_node/state_node.go

```diff
@@ -98,18 +100,23 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	events, err := m.prometheus.GetProcessedMetrics(mapping)
-	if err == nil {
-		m.enricher.Enrich(events)
+	if err != nil {
+		logger.Error(err)
+		reporter.Error(err)
+		return
 	}
 
-	return events, err
+	m.enricher.Enrich(events)
+	for _, event := range events {
+		reporter.Event(mb.Event{MetricSetFields: event})
```

Comment:
If we just add the event to the `MetricSetFields` here I think it will be reported under `kubernetes.state_node` but looking at the existing `data.json` it's under `kubernetes.node`:

```
  "kubernetes": {
    "node": {
      "cpu": {
        "allocatable": {
          "cores": 2
        },
```

We need to respect the namespace.

@sayden Can you verify that?